internal/errors/error.go

```diff
@@ -32,13 +32,13 @@ type Err struct {
 	Wrapped error
 }
 
-// New creates a new Err and supports the options of:
+// E creates a new Err with provided code and supports the options of:
 // WithOp - allows you to specify an optional Op (operation)
 // WithMsg() - allows you to specify an optional error msg, if the default
 // msg for the error Code is not sufficient.
 // WithWrap() - allows you to specify
 // an error to wrap
-func New(c Code, opt ...Option) error {
+func E(c Code, opt ...Option) error {
```

Comment:
I think E shouldn't have a required Code parameter. If you are wanting to wrap an error (but with advanced options) then you may not have a new code to provide. If there isn't another error given to wrap, then this should simply fill in a suitable default.

Reply:
Would you imagine there being a new `errors.WithCode` option so the caller can provide a code when calling `E` or just have it be the wrapped error code if provided otherwise `Unknown`?

Reply:
Yeah basically I think either you'd need WithWrap, or you'd need WithCode. Since we don't want to error from an error function, the code, if not specified, could be a default. But it's not strictly a required parameter given that if you wrap an error with this function it already has a code.

Reply:
👍